### ISSUE TITLE
Support for newline-delimited SNBT?

### ISSUE THREAD

**Issue by Pyroglyph** [janispritzkau/deno-minecraft#2]
FTB Quests stores quest data in SNBT, but it seems to be missing commas, instead being delimited with newlines.

<details>
<summary>Here's an example of a quest file</summary>

```snbt
{
	default_hide_dependency_lines: false
	default_quest_shape: ""
	filename: "miscellaneous"
	group: ""
	id: "300660B512C21504"
	order_index: 0
	quest_links: [ ]
	quests: [{
		description: ["Visit the origin point of the world."]
		id: "1BBA364E90171F30"
		subtitle: "Where it all started..."
		tasks: [{
			dimension: "minecraft:overworld"
			id: "2F7F0902B9C8A661"
			ignore_dimension: false
			position: [I;
				0
				0
				0
			]
			size: [I;
				1
				1
				1
			]
			title: "Origins"
			type: "location"
		}]
		x: 0.0d
		y: 0.0d
	}]
	subtitle: ["Extra quests, just for fun!"]
	title: "Miscellaneous"
}
```

</details>

Could the `parse` function be extended to also allow using a newline as a delimiter rather than requiring commas?
Or if you'd rather keep `parse` as a by-the-book implementation, maybe some kind of helper could be written that attempts to fix SNBT files that are malformed in this way.

**Comment by Pyroglyph** [comment]
I tried to throw together a helper for adding commas to newline-delimited SNBT.

```ts
snbtString
	.trim()
	.split(/\r?\n/)
	.map(line => line.match(/[\[\{\;]$/) ? line : line + ',')
	.join('\n')
	.replace(/,(\n?\s*[\]\}])/g, '$1')
	.replace(/,$/, '\n')
```

Or in plain English:
1. Trim any whitespace from the SNBT string
2. Split the SNBT string into lines
3. Append a comma to every line that doesn't end with a `[`, `{`, or `;`
4. Join all the lines back together into a single string
5. Replace any instances of commas preceding a closing bracket/brace (ignoring the newline and whitespace)
6. Remove the `,` after the final closing `}` and add a final newline

Running this on the sample quest I gave allows this library to successfully parse it, along with another larger quest file I created. I hope it's helpful, should you decide to add this feature.
